src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -367,12 +367,14 @@ BoundaryConditionCoupled(2, (:j, :i_backwards, :end), Float64)
 mutable struct BoundaryConditionCoupled{NDIMS, NDIMST2M1, uEltype <: Real, Indices}
     # NDIMST2M1 == NDIMS * 2 - 1
     # Buffer for boundary values: [variable, nodes_i, nodes_j, cell_i, cell_j]
-    u_boundary        :: Array{uEltype, NDIMST2M1} # NDIMS * 2 - 1
-    other_semi_index  :: Int
-    other_orientation :: Int
-    indices           :: Indices
-
-    function BoundaryConditionCoupled(other_semi_index, indices, uEltype)
+    u_boundary::Array{uEltype, NDIMST2M1} # NDIMS * 2 - 1
+    other_semi_index::Int
+    other_orientation::Int
+    indices::Indices
+    coupling_converter::Function
```

Comment:
`Function` is an abstract type, so it will result in some type instabilities later on. You should check whether you want to specialize it here.

Reply:
What about defining a type CouplingFunction? That might do the trick?

Reply:
Or just make it a parametric type to enforce specialization

Reply:
It seems that Julia does not subtype Function, or maybe I was just not able to.
Specialization could happen using the two equations involved. However, I would need to keep open the possibility of defining any function for one pair of equations.

Reply:
```julia
julia> struct Foo
           f::Function
       end

julia> call(foo, arg) = foo.f(arg)
call (generic function with 1 method)

julia> foo = Foo(identity)
Foo(identity)

julia> @code_warntype call(foo, 1)
MethodInstance for call(::Foo, ::Int64)
  from call(foo, arg) @ Main REPL[2]:1
Arguments
  #self#::Core.Const(call)
  foo::Foo
  arg::Int64
Body::Any
1 ─ %1 = Base.getproperty(foo, :f)::Function
│   %2 = (%1)(arg)::Any
└──      return %2


julia> struct FooSpecialzied{F}
           f::F
       end

julia> foo = FooSpecialzied(identity)
FooSpecialzied{typeof(identity)}(identity)

julia> @code_warntype call(foo, 1)
MethodInstance for call(::FooSpecialzied{typeof(identity)}, ::Int64)
  from call(foo, arg) @ Main REPL[2]:1
Arguments
  #self#::Core.Const(call)
  foo::Core.Const(FooSpecialzied{typeof(identity)}(identity))
  arg::Int64
Body::Int64
1 ─ %1 = Base.getproperty(foo, :f)::Core.Const(identity)
│   %2 = (%1)(arg)::Int64
└──      return %2
```

Reply:
That's what I meant - introduce a type parameter as in the second version

Reply:
OK, I think I implemented a specialization in that spirit.